test/jdk/java/nio/channels/DatagramChannel/AfterDisconnect.java

```diff
@@ -54,37 +56,75 @@
+        BindExceptionOnDisconnect(BindException x) {
+            super(x.getMessage());
+            initCause(x);
+        }
+    }
+
     @Test
     public void execute() throws IOException {
         IPSupport.throwSkippedExceptionIfNonOperational();
         boolean preferIPv6 = Boolean.getBoolean("java.net.preferIPv6Addresses");
         InetAddress lb = InetAddress.getLoopbackAddress();
 
         // test with default protocol family
-        try (DatagramChannel dc = DatagramChannel.open()) {
-            System.out.println("Test with default");
-            dc.bind(new InetSocketAddress(lb, 0));
-            test(dc);
-            test(dc);
-        }
-
-        // test with IPv6 socket
-        if (IPSupport.hasIPv6()) {
-            System.out.println("Test with IPv6 socket");
-            try (DatagramChannel dc = DatagramChannel.open(StandardProtocolFamily.INET6)) {
+        System.out.println("Test with default");
+        retry(() -> {
+            try (DatagramChannel dc = DatagramChannel.open()) {
                 dc.bind(new InetSocketAddress(lb, 0));
                 test(dc);
                 test(dc);
             }
+        }, BindExceptionOnDisconnect.class::isInstance, 5);
+
+        // test with IPv6 socket
+        if (IPSupport.hasIPv6()) {
+            System.out.println("Test with IPv6 socket");
+            retry(() -> {
+                try (DatagramChannel dc = DatagramChannel.open(StandardProtocolFamily.INET6)) {
+                    dc.bind(new InetSocketAddress(lb, 0));
+                    test(dc);
+                    test(dc);
+                }
+            }, BindExceptionOnDisconnect.class::isInstance, 5);
         }
 
         // test with IPv4 socket
         if (IPSupport.hasIPv4() && !preferIPv6) {
             System.out.println("Test with IPv4 socket");
-            try (DatagramChannel dc = DatagramChannel.open(StandardProtocolFamily.INET)) {
-                dc.bind(new InetSocketAddress(lb, 0));
-                test(dc);
-                test(dc);
+            retry(() -> {
+                try (DatagramChannel dc = DatagramChannel.open(StandardProtocolFamily.INET)) {
+                    dc.bind(new InetSocketAddress(lb, 0));
+                    test(dc);
+                    test(dc);
+                }
+            }, BindExceptionOnDisconnect.class::isInstance, 5);
+        }
+    }
+
+    // retry the given lambda (RetryableTest) if an exception
+    // that satisfies the predicate (retryOn) is caught.
+    <T extends Exception> void retry(RetryableTest<T> test,
+                                     Predicate<Throwable> retryOn,
+                                     int max) throws T {
+        for (int i=0; i < max; i++) {
+            try {
+                test.runTest();
+                break;
+            } catch (Throwable t) {
+                if (i < max -1 && retryOn.test(t)) {
+                    System.out.println("Got " + t + "; will retry");
+                } else throw t;
```

Comment:
What you would think about renaming this to "testWithRetry" so that it's clearer what it does? It could move up to the be the interface declarations so it is easier to find.